src/platform/windows/mod.rs

```diff
@@ -0,0 +1,1482 @@
+        }
+    }
+}
+
+impl Deref for OsIpcSharedMemory {
+    type Target = [u8];
+
+    #[inline]
+    fn deref(&self) -> &[u8] {
+        assert!(!self.ptr.is_null() && self.handle.is_valid());
+        unsafe {
+            slice::from_raw_parts(self.ptr, self.length)
+        }
+    }
+}
+
+impl OsIpcSharedMemory {
+    #[allow(exceeding_bitshifts)]
+    fn new(length: usize) -> Result<OsIpcSharedMemory,WinError> {
+        unsafe {
+            let (lhigh, llow) = if cfg!(target_pointer_width = "64") {
+                ((length >> 32) as u32, (length & 0xffffffffusize) as u32)
+            } else {
+                (0, length as u32)
+            };
+            let handle =
+                kernel32::CreateFileMappingA(INVALID_HANDLE_VALUE,
+                                             ptr::null_mut(),
+                                             winapi::PAGE_READWRITE | winapi::SEC_COMMIT,
+                                             lhigh, llow,
+                                             ptr::null_mut());
+            if handle == INVALID_HANDLE_VALUE {
+                return Err(WinError::last("CreateFileMapping"));
+            }
+
+            OsIpcSharedMemory::from_handle(handle, length)
+        }
+    }
+
+    // There is no easy way to query the size of the mapping -- you
+    // can use NtQuerySection, but that's an undocumented NT kernel
+    // API.  Instead we'll just always pass the length along.
+    //
+    // This function takes ownership of the handle, and will close it
+    // when finished.
+    fn from_handle(handle_raw: HANDLE, length: usize) -> Result<OsIpcSharedMemory,WinError> {
+        // turn this into a WinHandle, because that will
+        // take care of closing it
+        let handle = WinHandle::new(handle_raw);
+        let address = unsafe {
+            kernel32::MapViewOfFile(handle_raw,
+                                    winapi::FILE_MAP_ALL_ACCESS,
+                                    0, 0, 0)
+        };
+        if address.is_null() {
+            return Err(WinError::last("MapViewOfFile"));
+        }
+
+        Ok(OsIpcSharedMemory {
+            handle: handle,
+            ptr: address as *mut u8,
+            length: length
+        })
+    }
+
+    pub fn from_byte(byte: u8, length: usize) -> OsIpcSharedMemory {
+        unsafe {
+            // panic if we can't create it
+            let mem = OsIpcSharedMemory::new(length).unwrap();
+            for element in slice::from_raw_parts_mut(mem.ptr, mem.length) {
+                *element = byte;
+            }
+            mem
+        }
+    }
+
+    pub fn from_bytes(bytes: &[u8]) -> OsIpcSharedMemory {
+        unsafe {
+            // panic if we can't create it
+            let mem = OsIpcSharedMemory::new(bytes.len()).unwrap();
+            ptr::copy_nonoverlapping(bytes.as_ptr(), mem.ptr, bytes.len());
+            mem
+        }
+    }
+}
+
+pub struct OsIpcOneShotServer {
+    receiver: RefCell<Option<OsIpcReceiver>>,
```

Comment:
Actually, it's not even necessary to `consume()` -- see https://github.com/servo/ipc-channel/pull/114